### PR TITLE
Update Heapster guide to comprise core metrics in Kubernetes as general topic

### PR DESCRIPTION
**Pull request by pipo02mix**
Towards https://github.com/giantswarm/giantswarm/issues/2656

The idea is present what are core metrics in a nutshell, how they work since the appearance of Metrics API and keep the info how they were managed before with Heapster

### CODE REVIEW

**Comment by pipo02mix** [comment]
Yes please feel free to modify it.

File: content/guides/kubernetes-heapster/index.md [hunk 1]
```diff
@@ -1,19 +1,37 @@
...
 type = "page"
 weight = 140
 tags = ["recipe"]
 +++
 
-# Adding Metrics with Heapster
+# Metrics in Kubernetes
 
-Heapster is a simple monitoring solution that supports Kubernetes and CoreOS natively. Although we usually prefer [Prometheus](/guides/kubernetes-prometheus/), currently you need Heapster for integrating metrics in [Kubernetes Dashboard](/guides/install-kubernetes-dashboard/) or running the [Horizontal Pod Autoscaler](http://kubernetes.io/docs/user-guide/horizontal-pod-autoscaling/).
+Getting core metrics like CPU and Memory usage of resources in your cluster is important not only for your own monitoring purposes, but also for extended functionality like horizpontal Pod autoscaling.
```
Comment (rossf7) [inline content/guides/kubernetes-heapster/index.md]: horizontal

File: content/guides/kubernetes-heapster/index.md [hunk 2]
```diff
@@ -1,19 +1,37 @@
 +++
-title = "Adding Metrics with Heapster"
-description = "Recipe to get the Heapster monitoring solution running."
-date = "2017-09-04"
+title = "Getting Basic Metrics in your Cluster"
+description = "Recipe to get enable a metrics monitoring solution running on your Kubernetes cluster."
```
Comment (rossf7) [inline content/guides/kubernetes-heapster/index.md]: "get enable"

File: content/guides/kubernetes-heapster/index.md [hunk 3]
```diff
@@ -1,19 +1,37 @@
...
+Getting core metrics like CPU and Memory usage of resources in your cluster is important not only for your own monitoring purposes, but also for extended functionality like horizpontal Pod autoscaling.
 
-You can deploy Heapster as a standalone solution (recommended if you are running Prometheus for monitoring anyway) or as a complete monitoring solution, e.g. together with InfluxDB and Grafana (currently not documented here).
+The kubelet exposes core metrics of the nodes/pods via an endpoint, however, you need an additional monitoring tool to collect and expose those metrics. Until recently, the tool of choice for this was Heapster and its API. But with the recent move to a more general Metrics API you get such metrics directly from the Kubernetes API endpoint. However, for this to work you need to still run a monitoring tool that collects and exposes the metrics as the Kubernetes API only aggregates monitoring backends to be exposed by it.
```
Comment (rossf7) [inline content/guides/kubernetes-heapster/index.md]: Nitpicky but I'd start a new sentence at "As the Kubernetes API".

File: content/guides/kubernetes-heapster/index.md [hunk 4]
```diff
@@ -1,19 +1,37 @@
...
 
-## Deploy Heapster (standalone)
+
+# Adding Metrics with metric server
+
+Since version 1.8, Kubernetes offers a new endpoint `/apis/metrics.k8s.io/` to get the core metrics. But there is no default backend for aggregating and exposing those.
```
Comment (rossf7) [inline content/guides/kubernetes-heapster/index.md]: exposing them

File: content/guides/kubernetes-heapster/index.md [hunk 5]
```diff
@@ -1,19 +1,37 @@
...
+
+Kubernetes community has worked refactoring old Heapster in order to make it implement new API. The project has been renamed as "[metric server](https://github.com/kubernetes-incubator/metrics-server)" and it can be deployed in the cluster applying the provided yaml resources
```
Comment (rossf7) [inline content/guides/kubernetes-heapster/index.md]: The K8s community

File: content/guides/kubernetes-heapster/index.md [hunk 6]
```diff
@@ -1,19 +1,37 @@
...
+
+```
+git clone https://github.com/kubernetes-incubator/metrics-server/tree/master/deploy/1.8%2B
+cd metric-server
+kubectl apply deploy/1.8+/
+```
+
+__Warning__: There is a requirement for deploying Metrics API backend, the [aggregation layer should be configured](https://kubernetes.io/docs/tasks/access-kubernetes-api/configure-aggregation-layer). The reason for that is the metric server is an extended Kubernetes API. 
+So once enabled Kubernetes api server will proxy requests reaching metrics endpoint to the aforementioned backend.
```
Comment (rossf7) [inline content/guides/kubernetes-heapster/index.md]: API

File: content/guides/kubernetes-heapster/index.md [hunk 7]
```diff
@@ -22,4 +40,16 @@ kubectl apply --filename https://raw.githubusercontent.com/kubernetes/heapster/m
 kubectl apply --filename https://raw.githubusercontent.com/kubernetes/heapster/master/deploy/kube-config/standalone/heapster-controller.yaml
 ```
 
-If you have the [Kubernetes Dashboard](/guides/install-kubernetes-dashboard/) installed, you should be able to see metrics being shown in the workload overview as well as on the pods after a reload. If they don't show up directly, just wait a moment and try again.
+# Use cases
+
+There are some common cases where core metrics are used by Kubernetes:
+
+- Horizontal Pod Autoscaler: It scale pods automatically based on CPU usage metrics or custom metrics (not explained here). More information [here](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale).
```
Comment (rossf7) [inline content/guides/kubernetes-heapster/index.md]: scales

File: content/guides/kubernetes-heapster/index.md (hunk 4 above)
Comment (puja108) [reply content/guides/kubernetes-heapster/index.md]: I rewrite this and everything below this, sorry for not saying its WIP

File: content/guides/kubernetes-heapster/index.md (hunk 5 above)
Comment (puja108) [reply content/guides/kubernetes-heapster/index.md]: s.o.

File: content/guides/kubernetes-heapster/index.md (hunk 6 above)
Comment (puja108) [reply content/guides/kubernetes-heapster/index.md]: s.o.

**Comment by puja108** [comment]
@rossf7 I now finished the whole rewrite, included your typo comments, maybe you wanna take another look? should be ready now, sorry for not saying it was WIP before

File: content/guides/kubernetes-heapster/index.md [hunk 8]
```diff
@@ -1,19 +1,35 @@
...
+The kubelet exposes core metrics of the nodes/pods via an endpoint, however, you need an additional monitoring tool to collect and expose those metrics. Until recently, the tool of choice for this was Heapster and its API. But with the recent move to a more general Metrics API you get such metrics directly from the Kubernetes API endpoint. However, for this to work you need to still run a monitoring tool that collects and exposes the metrics, as the Kubernetes API only aggregates monitoring backends to be exposed by it.
 
-## Deploy Heapster (standalone)
+# Adding Metrics with Metrics Server
+
+[Metrics Server](https://github.com/kubernetes-incubator/metrics-server) is a cluster-wide aggregator of resource usage data, which it collects from the Summary API of the kubelets of each node. It registeres itself in the main API server through Kubernetes Aggregator and thus is discoverable through the same API endpoint as the rest of Kubernetes under `/apis/metrics.k8s.io/`.
```
Comment (rossf7) [inline content/guides/kubernetes-heapster/index.md]: registers

File: content/guides/kubernetes-heapster/index.md [hunk 9]
```diff
@@ -22,4 +38,16 @@ kubectl apply --filename https://raw.githubusercontent.com/kubernetes/heapster/m
...
+There are some common cases where Core Metrics are used by Kubernetes:
+
+- Horizontal Pod Autoscaler: It scales pods automatically based on CPU or custom metrics (not explained here). More information [here](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale).
+- Kubectl top: The command `top` of our beloved Kubernetes CLI display metrics directly in the terminal.
+- Kubernetes dashboard: See Pod and Nodes metrics integrated into the main Kubernetes UI dashboard. More info [here](/guides/install-kubernetes-dashboard)
+- Scheduler: In the future Core Metrics will be considered in order to schedule best-effort Pods.
+
+# Further Reading
+
+- [Kuberentes Monitoring Architecture](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/monitoring_architecture.md)
```
Comment (rossf7) [inline content/guides/kubernetes-heapster/index.md]: Kubernetes

File: content/guides/kubernetes-heapster/index.md [hunk 10]
```diff
@@ -1,19 +1,35 @@
...
+
+To deploy it run:
+
+```bash
+git clone https://github.com/kubernetes-incubator/metrics-server/tree/master/deploy/1.8%2B
+cd metrics-server
+kubectl apply --filename deploy/1.8+/
+```
+
+__Note__: For Metrics Server to work you need to have Kubernetes API Aggregator enabled on your cluster. This is enabled by default on clusters started after February 14th 2018. For older clusters you can use Heapster as described below.
```
Comment (rossf7) [inline content/guides/kubernetes-heapster/index.md]: <3 Good to mention this and really clear whether your cluster has it.

Comment (puja108) [reply content/guides/kubernetes-heapster/index.md]: took me a bit to find out when we released it

**Pull request merged**